src/lib/pci_virtio_net_vpnkit.c

```diff
@@ -108,6 +118,15 @@ struct vif_info {
 	uint8_t mac[6];
 } __packed;
 
+struct disconnect_reason {
+	uint8_t len;
+	char msg[256];
+} __packed;
+
+struct msg_response {
```

Comment:
I would move this declaration above `disconnect_reason` as this is what we always get. and then `disconnect_reason` is would be in data.

Also, it might make sense to make this a union where it's either `data` or `disconnect_reason` with `response_type` the selector.